File: Data Structures & Algorithms/insert-new-interval/submission-2.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        //Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        List<int[]> res = new ArrayList<>();
        int[] pre = newInterval;
        for (int i = 0; i < intervals.length; i++) {
            int[] cur = intervals[i];

            if (pre[1] < cur[0]) {
                res.add(pre);
                pre = cur;
            }
            else if (pre[0] > cur[1]) {
                res.add(cur);
            }
            else {
                pre[0] = Math.min(pre[0], cur[0]);
                pre[1] = Math.max(pre[1], cur[1]);
            }
        }
        res.add(pre);
        return res.toArray(new int[res.size()][]);
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-2.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        start, end = newInterval
        res = []
        for cur_start, cur_end in intervals:
            if cur_end < start:
                res.append([cur_start, cur_end])
                print(1)
            elif cur_start <= start <= cur_end or cur_start <= end <= cur_end:
                start = min(cur_start, start)
                end = max(cur_end , end)
                print(2)
            elif cur_start > end:
                res.append([start, end])
                start, end = cur_start, cur_end
                print(3)
        res.append([start, end])
        return res